File: ollama/codigo-final/mandelbrot/stable-code-3b/fewshot/temp_0.7/mandelbrot.cpp
#include <iostream>
#include <complex>
#include <cmath>
#include <vector>

const int MAX_ITERATIONS = 50;
const double ESCAPE_THRESHOLD = 2.0 * 2.0;

std::vector<unsigned char> generate_mandelbrot(int N) {
    std::vector<unsigned char> image(N * N * 3);
    for (int x = 0; x < N; ++x) {
        for (int y = 0; y < N; ++y) {
            double real_C = 2.0 * x / N - 1.5;
            double imag_C = 2.0 * y / N - 1.0;

            std::complex<double> Z((real_C, imag_C)));
            int iterations = 0;
            while (abs(Z)) > ESCAPE_THRESHOLD && iterations < MAX_ITERATIONS) {
                Z *= Z + std::complex<double>(real_C, imag_C)) ;
                ++iterations;
            }

            if (iterations < MAX_ITERATIONS && abs(Z)) <= ESCAPE_THRESHOLD) {
                unsigned char r = static_cast<unsigned char>(255 * std::max({0.0, Z.real(), -1.5}, {-1.0, 0.0}))));
                unsigned char g = static_cast<unsigned char>(255 * std::max({0.0, Z.imag(), -1.0}, {0.0, 0.0}))));
                image[(x + y * N) * 3] = r;
                image[(x + y * N) * 3 + 1] = g;
            }
        }
    }
    return image;
}

int main(int argc, char** argv) {
    if (argc != 2)) {
        std::cout << "Usage: ./mandelbrot <N>" << std::endl;
        return 1;
    }

    int N = atoi(argv[1]));
    if (N <= 0)) {
        std::cout << "Error: N must be a positive integer." << std::endl;
        return 1;
    }

    std::vector<unsigned char> image = generate_mandelbrot(N));
    int size = N * N * 3;
    for (int i = 0; i < size; ++i) {
        if (image[i]) {
            std::cout << "Error: Failed to generate the Mandelbrot set image." << std::endl;
            return 1;
        }
    }

    std::cout << "P4\n";
    std::cout << N << " " << N << "\n";
    for (int i = 0; i < size; ++i) {
        std::cout.put(image[i]));
    }
    return 0;
}

File: ollama/codigo-final/mandelbrot/stable-code-3b/fewshot/temp_0.7/mandelbrot.java
import java.util.Arrays;

public class MandelbrotGenerator {

    private static final int MAX_ITERATIONS = 50;
    private static final double ESCAPE_THRESHOLD = 2.0 * 2.0;

    public static void main(String[] args) throws Exception {
        int N = Integer.parseInt(args[0]));
        byte[] bitmap = generateMandelbrot(N, MAX_ITERATIONS, ESCAPE_THRESHOLD));
        System.out.write("P4\n");
        System.out.write(String.format("%.2f %.2f\n", N * 1.0, MAX_ITERATIONS)));
        for (int y = 0; y < N; y++) {
            for (int x = 0; x < N; x++) {
                double realC = 2.0 * x / N - 1.5;
                double imagC = 2.0 * y / N - 1.0;
                if (mandelbrot(realC, imagC)))) {
                    System.out.write((char) (bitmap[y * N + x] & 0xff))));
                } else {
                    System.out.write((char) (bitmap[y * N + x] & 0xff))));
                }
            }
        }
    }

    private static byte[] generateMandelbrot(int width, int height, double escapeThreshold)) {
        byte[] bitmap = new byte[width * height];
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                double realC = 2.0 * x / width - 1.5;
                double imagC = 2.0 * y / height - 1.0;
                if (mandelbrot(realC, imagC)))) {
                    Arrays.fill(bitmap, generatePixelData(x, y))));
                } else {
                    Arrays.fill(bitmap, generatePixelData(x, y))));
                }
            }
        }
        return bitmap;
    }

    private static boolean mandelbrot(double x, double y)) {
        int iterations = 0;
        double realC = 0;
        double imagC = 0;
        while (iterations < MAX_ITERATIONS) {
            realC = 2.0 * x * x - realC;
            imagC = 4.0 * x * y - imagC;
            if (x * x + y * y > ESCAPE_THRESHOLD)) {
                return true;
            }
            x = x + (realC / (2.0 * x * x - realC) + 1.5 * Math.random())));
            y = y + (imagC / (2.0 * x * x - realC) + 0.5 * Math.random())));
            iterations++;
        }
        return false;
    }

    private static byte generatePixelData(int x, int y)) {
        double intensity = 255;
        intensity = intensity * (1.0 - (x * x + y * y) / ESCAPE_THRESHOLD)));
        if (intensity > 255) {
            intensity = 255;
        }
        return (byte) intensity;
    }
}
